Core/Dialog/PSPSaveDialog.cpp

```diff
@@ -640,7 +640,7 @@ int PSPSaveDialog::Update(int animSpeed)
 	// Check if it has changed, reload it.
 	// TODO: Cut down on preloading?  This rebuilds the list from scratch.
 	int size = Memory::Read_U32(requestAddr);
-	if (memcmp(Memory::GetPointer(requestAddr), &originalRequest, size) != 0) {
+	if (memcmp(Memory::GetPointerRange(requestAddr, size), &originalRequest, size) != 0) {
```

Comment:
A problem with this is that GetPointerRange can return nullptr, and if the compiler deduces that, we're introducing undefined behavior which can cause strange and bad compiler "optimizations".

I really think we need to separate range checking (Memory::IsValidRange) and GetPointer in any situation where we don't naturally null check.

Maybe GetPointer(Range)(Write) is a bad idea and we should always do IsValidRange and then GetPointer*Unchecked...

Reply:
Well, I'd rather have it log or etc. and return a clear nullptr than just die in outerspace.  So I still think this is an improvement.

I agree size checks/null checks should be made, though.

-[Unknown]